### PR TITLE
TEMP: testing if adding --stack-yaml addresses build fail on windows

### PR DESCRIPTION
**Pull request by yarikoptic**
ref: https://github.com/datalad/git-annex/issues/168

### CODE REVIEW

**Comment by jkniiv** [comment]
For best effect I think you need to address the following lines as well with the added `--stack-yaml` option or the proper stack.yaml file:
  - https://github.com/datalad/git-annex/blob/cfe87684e57758f16b88ca572ea5967132917891/.github/workflows/build-windows.yaml#L113 (stack setup)
  - https://github.com/datalad/git-annex/blob/cfe87684e57758f16b88ca572ea5967132917891/.github/workflows/build-windows.yaml#L121 (stack cache)
  - https://github.com/datalad/git-annex/blob/cfe87684e57758f16b88ca572ea5967132917891/.github/workflows/build-windows.yaml#L126 (dependencies build)
  - https://github.com/datalad/git-annex/blob/cfe87684e57758f16b88ca572ea5967132917891/.github/workflows/build-windows.yaml#L131 (version info)
  - https://github.com/datalad/git-annex/blob/cfe87684e57758f16b88ca572ea5967132917891/.github/workflows/build-windows.yaml#L139 (installer build)

Basically wherever there is `stack` it should be replaced with `stack --stack-yaml stack-lts-18.13.yaml` because with Haskell Stack the whole build environment is ultimately defined by the stack.yaml. For my personal Windows builds I use a Justfile (just is a modern make-like utility) and there I defined a (template-style) variable called `stack-cmd` which is now effectively the same as `stack --stack-yaml stack-lts-18.13.yaml` (actually it takes the option from an environment variable but that's just an small detail). So I replaced all invocations of `stack` with `{{ stack-cmd }}` and that produced a working binary from version 10.20230802 sources via `{{ stack-cmd }} setup && {{ stack-cmd }} build`. Maybe something similar can be done here with the github workflow definition?

**Comment by yarikoptic** [comment]
Thank you @jkniiv ! Sounds sensible.  But as I think this is all temporary (right @joeyh ) let me try other approach -- just rename the damn config file into stack.yaml and be done ;)

**Comment by yarikoptic** [comment]
ok, that simple way worked, I will stick to it for now

**Comment by yarikoptic** [comment]
I do not expect surprises, so will merge
